presto-connector-tests/src/main/java/com/facebook/presto/connector/meta/SupportedTestCondition.java

```diff
@@ -0,0 +1,139 @@
+ *
+ * ConnectorTest1.testN has supported features {FEATURE1, FEATURE2, FEATURE3}
+ * ConnectorTest2.testN has supported features {FEATURE1, FEATURE2}
+ *
+ * Required features are gathered from the test method, its declaring class,
+ * and all of the super classes of its declaring class, recursively .
+ *
+ * ConnectorTestN.test0 has required features {FEATURE1}
+ * ConnectorTestN.test1 has required features {FEATURE1, FEATURE2}
+ * ConnectorTestN.test2 has required features {FEATURE1, FEATURE2, FEATURE3}
+ *
+ * A test is disabled if the test method has required features not contained in
+ * its supported features.
+ *
+ * Gathering required features for tests in inner classes is not supported.
+ * Feel free to add support if needed.
+ *
+ * Test interfaces can extend multiple other test interfaces. Required features
+ * are correctly gathered from all extended interfaces.
+ *
+ * BaseTest1 @RequiredFeatures({FEATURE1})     BaseTest2 @RequiredFeatures({FEATURE2})
+ *                                         \ /
+ *                                       SubTest
+ *                                           testMethod()
+ *
+ * The required features of SubTest.testMethod are {FEATURE1, FEATURE2}
+ */
+public class SupportedTestCondition
+        implements ExecutionCondition
+{
+    @Override
+    public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext extensionContext)
+    {
+        Optional<Method> testMethod = extensionContext.getTestMethod();
+
+        if (!testMethod.isPresent()) {
+            return enabled("");
+        }
+
+        Class<?> testClass = extensionContext.getRequiredTestClass();
+
+        Optional<RequiredFeatures> methodRequires = testMethod.map(method -> method.getAnnotation(RequiredFeatures.class));
+        List<RequiredFeatures> classRequires = getClassRequiredFeatures(testMethod.get().getDeclaringClass());
+        Optional<SupportedFeatures> classSupports = Optional.ofNullable(testClass.getAnnotation(SupportedFeatures.class));
+
+        Set<ConnectorFeature> requiredFeatures = Stream.concat(
+                streamOfRequired(methodRequires),
+                classRequires.stream().flatMap(r -> Arrays.stream(r.value())))
+                .distinct()
+                .collect(Collectors.toCollection(HashSet::new));
+
+        Set<ConnectorFeature> supportedFeatures = streamOfSupported(classSupports)
+                .distinct()
+                .collect(toImmutableSet());
+
+        requiredFeatures.removeAll(supportedFeatures);
+
+        return requiredFeatures.isEmpty() ?
+                enabled("All required features present") :
+                disabled("Missing required features: " + Joiner.on(", ").join(requiredFeatures));
+    }
+
+    private static List<RequiredFeatures> getClassRequiredFeatures(Class<?> testClass)
+    {
+        assertNull(testClass.getEnclosingClass(), "Add support (and tests!) if you need it");
+        Class<?>[] interfaces = testClass.getInterfaces();
```

Comment:
Because putting the test methods in interfaces is apparently the JUnit 5 way and I haven't yet run into a scenario where I wanted or needed to put test methods in a class. I'm OK with adding it if we need it, but for now it seems like YAGNI unless you have a specific use in mind.